daemon/k8s_watcher.go

```diff
@@ -1306,13 +1329,61 @@ func updateCNPNodeStatus(ciliumNPClient clientset.Interface, cnp *cilium_v2.Cili
 		}
 	}
 
-	cnp.SetPolicyStatus(nodeName, cnpns)
 	ns := k8sUtils.ExtractNamespace(&cnp.ObjectMeta)
 
 	switch {
 	case ciliumUpdateStatusVerConstr.Check(k8sServerVer):
-		_, err2 = ciliumNPClient.CiliumV2().CiliumNetworkPolicies(ns).UpdateStatus(cnp)
+		// This is a JSON Patch used to create the `/status/nodes` field in the
+		// CNP. If we don't create, replacing the status for this node will
+		// fail as the path does not exist.
+		// Worst case scenario is that all nodes try to perform this operation
+		// and only one node will succeed. This can be moved to the
+		// cilium-operator which will create the path for all nodes. However
+		// performance tests have shown that performing 2 API calls to
+		// kube-apiserver for 1000 nodes versus performing 1 API call, where
+		// one of the nodes would "create" the `/status` path before all other
+		// nodes tried to replace their own status resulted in a gain of X %
+		createStatusAndNodePatch := []jsonPatch{
+			{
+				OP:    "test",
+				Path:  "/status",
+				Value: nil,
+			},
+			{
+				OP:   "add",
+				Path: "/status",
+				Value: cilium_v2.CiliumNetworkPolicyStatus{
+					Nodes: map[string]cilium_v2.CiliumNetworkPolicyNodeStatus{
+						nodeName: cnpns,
+					},
+				},
+			},
+		}
+
+		createStatusAndNodePatchJSON, err := json.Marshal(createStatusAndNodePatch)
+		if err != nil {
+			return err
+		}
+
+		_, err2 = ciliumNPClient.CiliumV2().CiliumNetworkPolicies(ns).Patch(cnp.GetName(), types.JSONPatchType, createStatusAndNodePatchJSON, "status")
```

Comment:
Please use more descriptive error names than `err` and `err2` that are more evocative of what types of errors they are. This makes the code more readable. In the past CNP status update code, a similar convention to naming errors was used and it was quite difficult to read the code.